sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import org.apache.spark.sql.types.StructType;
+
+import java.util.Map;
+
+/**
+ * Record class that holds view information.
+ *
+ * @param ident the view identifier
+ * @param sql the SQL text that defines the view
+ * @param currentCatalog the current catalog
+ * @param currentNamespace the current namespace
+ * @param schema the view query output schema
+ * @param queryColumnNames the query column names
+ * @param columnAliases the column aliases
+ * @param columnComments the column comments
+ * @param properties the view properties
+ */
+public record ViewInfo(Identifier ident,
```

Comment:
can we use two space indentation?

Reply:
done

Reply:
it's not done... it should be
```
public record ViewInfo(
  Identifier ident,
  ...
```

Reply:
We have the `ColumnDefaultValue` API in ds v2 and it's just plain java class

Reply:
sorry I misunderstood the first comment. Indeed, no need to use `record` here, so I've switched to a plain java class